File: MDVision.IEHR/Controls/Admin/Admin_BasicFeeGroup_Detail.js
basicfeegroupDetail = {
    params: [],
    Enable: false,
    Load: function (params) {
        basicfeegroupDetail.params = params;
        
        //initialization of date-picker.
        utility.CreateDatePicker('tblbasicfeegroupDetail #txtEndDate', function () {
            //on-change callback method 
        });

        var self = null;
        if (basicfeegroupDetail.params.PanelID == "basicfeegroupDetail")
            self = $('#basicfeegroupDetail');
        else
            self = $('#' + basicfeegroupDetail.params.PanelID + ' #basicfeegroupDetail');

        if (globalAppdata.AppUserName.toLowerCase() != "mdvision") {
            self.find("#ddlEntity").attr('disabled', 'disabled');
        }
        self.loadDropDowns(true).done(function () {            
            //if (globalAppdata['IsAdmin'] != "True") {
            //    $("#basicfeegroupDetail #divBasicFeeGroup_Entity").css("display", "none");
            //    $("#basicfeegroupDetail #ddlEntity").val(globalAppdata["SeletedEntityId"]);
                
            //}
            if (globalAppdata.AppUserName.toLowerCase() != "mdvision") {
                self.find("#ddlEntity").val(globalAppdata["SeletedEntityId"]);
            }
            $("#basicfeegroupDetail #chkIsActive").attr("checked", true);
            basicfeegroupDetail.LoadBasicFeeGroup();
        });

        
           
    },

    LoadBasicFeeGroup: function () {
        if (basicfeegroupDetail.params.mode == "Add") {

            //serialize data
            $('#frmBasicFeeGroupDetail').data('serialize', $('#frmBasicFeeGroupDetail').serialize());
            basicfeegroupDetail.ValidationBasicFeeGroup();
        }
        else if (basicfeegroupDetail.params.mode == "Edit") {
            basicfeegroupDetail.FillBasicFeeGroup(basicfeegroupDetail.params.BasicFeeGroupId).done(function (response) {
                if (response.status != false) {
                    var basicfeegroup_detail = JSON.parse(response.BasicFeeGroupFill_JSON);
                    var self = $("#basicfeegroupDetail");
                    utility.bindMyJSON(true, basicfeegroup_detail, false, self).done(function () {

                        if (basicfeegroup_detail.ChkIsActive == 'True')
                            $("#basicfeegroupDetail #chkIsActive").attr("checked", true);
                        else
                            $("#basicfeegroupDetail #chkIsActive").attr("checked", false);
                        basicfeegroupDetail.LoadNextBasicFeeGroup(basicfeegroup_detail.ddlEntity, basicfeegroup_detail.ddlNextBasicFeeGroup);
                        basicfeegroupDetail.ValidationBasicFeeGroup();

                        //serialize data
                        $('#frmBasicFeeGroupDetail').data('serialize', $('#frmBasicFeeGroupDetail').serialize());

                    });

                   
                }
                else {
                    utility.DisplayMessages(response.Message, 3);
                }
            });
        }
    },

    BasicFeeGroupSave: function () {
        var strMessage = "";
        var self = $("#basicfeegroupDetail");
        var myJSON = self.getMyJSON();
        if (basicfeegroupDetail.params.mode == "Add") {
            AppPrivileges.GetFormPrivileges("Basic Fee Group", "ADD", "FORM_PRIVILEGE", "GET_FORM_PRIVILEGE", function (strMessage) {
                if (strMessage == "") {
                    basicfeegroupDetail.SaveBasicFeeGroup(myJSON).done(function (response) {
                        if (response.status != false) {
                            Admin_BasicFeeGroup.BasicFeeGroupSearch(response.BasicFeeGroupId);
                            utility.DisplayMessages(response.message, 1);
                            CacheManager.BindCodes('GetBasicFeeGroup', true);
                            UnloadActionPan();
                        }
                        else {
                            utility.DisplayMessages(response.Message, 3);
                        }
                    });
                }
                else
                    utility.DisplayMessages(strMessage, 2);
            });
        }
        else if (basicfeegroupDetail.params.mode == "Edit") {
            AppPrivileges.GetFormPrivileges("Basic Fee Group", "EDIT", "FORM_PRIVILEGE", "GET_FORM_PRIVILEGE", function (strMessage) {
                if (strMessage == "") {
                    basicfeegroupDetail.UpdateBasicFeeGroup(myJSON, basicfeegroupDetail.params.BasicFeeGroupId).done(function (response) {
                        if (response.status != false) {
                            Admin_BasicFeeGroup.BasicFeeGroupSearch(basicfeegroupDetail.params.BasicFeeGroupId);
                            utility.DisplayMessages(response.message, 1);
                            CacheManager.BindCodes('GetBasicFeeGroup', true);
                            UnloadActionPan();
                        }
                        else {
                            utility.DisplayMessages(response.Message, 3);
                        }
                    });
                }
                else
                    utility.DisplayMessages(strMessage, 2);
            });
        }
    },

    ValidationBasicFeeGroup: function () {
        $('#frmBasicFeeGroupDetail')
          .bootstrapValidator({
              live: 'disabled',
              message: 'This value is not valid',
              feedbackIcons: {
                  valid: 'glyphicon glyphicon-ok',
                  invalid: 'glyphicon glyphicon-remove',
                  validating: 'glyphicon glyphicon-refresh'
              },
              fields: {
                  Name: {
                      group: '.col-sm-4',
                      validators: {
                          notEmpty: {
                              message: ''
                          },
                      }
                  },
                  Entity: {
                      group: '.col-sm-4',
                      validators: {
                          notEmpty: {
                              message: ''
                          },
                      }
                  },
              }
          })
       .on('success.form.bv', function (e) {
           e.preventDefault();
           basicfeegroupDetail.BasicFeeGroupSave();
       });
    },

    LoadNextBasicFeeGroup: function (entityID, ddlNextBasicFeeGroup) {
        var EntityID = "";
        basicfeegroupDetail.FillNextBasicFeeGroup(entityID).done(function (response) {
            if (response.status != false) {
                var basicfeegroup_detail = JSON.parse(response.NextBasicFeeGroupLoad_JSON);
                $("#basicfeegroupDetail #ddlNextBasicFeeGroup").empty();
                $("#basicfeegroupDetail #ddlNextBasicFeeGroup").append($('<option/>', {
                    value: "",
                    html: "- SELECT -"
                }));
                $.each(basicfeegroup_detail, function (i, item) {
                    $("#basicfeegroupDetail #ddlNextBasicFeeGroup").append(
                        $('<option/>', {
                            value: item.BasicFeeGroupId,
                            html: item.EntityName +" - "+ item.ShortName
                        })
                    );
                });
            }
            $('#basicfeegroupDetail #ddlNextBasicFeeGroup').val(ddlNextBasicFeeGroup);
        });
    },

    SaveBasicFeeGroup: function (BasicFeeGroupData) {
        var data = "BasicFeeGroupData=" + BasicFeeGroupData;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_BASIC_FEE_GROUP_DETAIL", "SAVE_BASIC_FEE_GROUP");
    },

    UpdateBasicFeeGroup: function (BasicFeeGroupData, BasicFeeGroupID) {
        var data = "BasicFeeGroupData=" + BasicFeeGroupData + "&BasicFeeGroupID=" + BasicFeeGroupID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_BASIC_FEE_GROUP_DETAIL", "UPDATE_BASIC_FEE_GROUP");
    },

    FillBasicFeeGroup: function (BasicFeeGroupID) {
        var data = "BasicFeeGroupID=" + BasicFeeGroupID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_BASIC_FEE_GROUP_DETAIL", "FILL_BASIC_FEE_GROUP");
    },

    FillNextBasicFeeGroup: function (EntityID) {
        var data = "EntityID=" + EntityID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_BASIC_FEE_GROUP_DETAIL", "FILL_NEXT_BASIC_FEE_GROUP");
    },

    UpdateBasicFeeGroupActiveInactive: function (BasicFeeGroupID, IsActive) {
        var data = "BasicFeeGroupID=" + BasicFeeGroupID + "&IsActive=" + IsActive;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_BASIC_FEE_GROUP_DETAIL", "UPDATE_BASIC_FEE_GROUP_ACTIVE_INACTIVE");
    },

    UnLoad: function () {

        utility.UnLoadDialog("frmBasicFeeGroupDetail", function () {
            UnloadActionPan();
        }, function () {
            UnloadActionPan();
        });
    },

    ShowHistory: function () {
        var PanelID = 'basicfeegroupDetail';
        var ParentCtrl = 'basicfeegroupDetail';
        var ProfileName = 'Basic Fee Group';
        var DBTableName = 'BasicFeeGroup';
        var ColumnKeyId = basicfeegroupDetail.params.BasicFeeGroupId;

        utility.ShowHistory(ParentCtrl, PanelID, ProfileName, DBTableName, ColumnKeyId);
    },
}

File: MDVision.IEHR/Controls/Admin/Admin_BasicFeeGroup_Detail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MDVision.Datasets;
using MDVision.Business.BCommon;
using System.Data;

using Newtonsoft.Json;
using MDVision.IEHR.Common;
using MDVision.Common.Utilities;
using MDVision.Common.Shared;
using MDVision.Business.BLL;

namespace MDVision.IEHR.Controls.Admin
{
    public class Admin_BasicFeeGroup_Detail
    {
        private BLLFeeSchedule BLLFeeScheduleObj = null;
        public Admin_BasicFeeGroup_Detail()
        {
            BLLFeeScheduleObj = new BLLFeeSchedule();
        }
        #region Singleton
        private static Admin_BasicFeeGroup_Detail _obj = null;
        public static Admin_BasicFeeGroup_Detail Instance()
        {
            if (_obj == null)
                _obj = new Admin_BasicFeeGroup_Detail();
            return _obj;
        }
        #endregion

        #region Private Functions
        /// <summary>
        /// Saves the Fee Group.
        /// </summary>
        /// <param name="fieldsJSON">The fields json.</param>
        /// <returns>Json string containing Succes message or Exception message</returns>
        private string SaveBasicFeeGroup(string fieldsJSON)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Basic Fee Group", "ADD")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    System.Web.Script.Serialization.JavaScriptSerializer ser = new System.Web.Script.Serialization.JavaScriptSerializer();
                    var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                    DSFeeSchedule dsBasicFeeGroup = new DSFeeSchedule();
                    DSFeeSchedule.BasicFeeGroupRow dr = dsBasicFeeGroup.BasicFeeGroup.NewBasicFeeGroupRow();

                    dr.ShortName = SearchedfieldsJSON["txtName"].Trim();
                    dr.Description = SearchedfieldsJSON["txtDescription"];
                    if (!string.IsNullOrEmpty(MDVUtility.ToStr(SearchedfieldsJSON["txtEndDate"])))
                        dr.EndDate = MDVUtility.ToDateTime(SearchedfieldsJSON["txtEndDate"]);
                    if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlNextBasicFeeGroup"]))
                        dr.NextBasicFeeGroupId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlNextBasicFeeGroup"]);

                    dr.IsActive = MDVUtility.ToStr(SearchedfieldsJSON["chkIsActive"]) == "True" ? true : false;

                    dr.CreatedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                    dr.CreatedOn = DateTime.Now;
                    dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                    dr.ModifiedOn = DateTime.Now;
                    if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlEntity"]))
                        dr.EntityId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlEntity"]);

                    #region Database Insertion
                    dsBasicFeeGroup.BasicFeeGroup.AddBasicFeeGroupRow(dr);
                    BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.InsertBasicFeeGroup(ref dsBasicFeeGroup);
                    dsBasicFeeGroup = obj.Data;
                    if (obj.Data != null)
                    {
                        var response = new
                        {
                            status = true,
                            message = Common.AppPrivileges.Save_Message,
                            BasicFeeGroupId = dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows[0][dsBasicFeeGroup.BasicFeeGroup.BasicFeeGroupIdColumn.ColumnName]
                        };
                        return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    #endregion
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Updates the BasicFeeGroup Link.
        /// </summary>
        /// <param name="fieldsJSON">The fields json.</param>
        /// <param name="ProcedureCategoryId">The BasicFeeGroup identifier.</param>
        /// <returns>Json string containing Succes message or Exception message</returns>
        private string UpdateBasicFeeGroup(string fieldsJSON, Int64 BasicFeeGroupId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Basic Fee Group", "EDIT")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    System.Web.Script.Serialization.JavaScriptSerializer ser = new System.Web.Script.Serialization.JavaScriptSerializer();
                    var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                    DSFeeSchedule dsBasicFeeGroup = new DSFeeSchedule();
                    //DSFeeSchedule.BasicFeeGroupRow dr = dsBasicFeeGroup.BasicFeeGroup.NewBasicFeeGroupRow();
                    BLObject<DSFeeSchedule> objLoad = BLLFeeScheduleObj.LoadBasicFeeGroup(BasicFeeGroupId, null, null, null, null);
                    dsBasicFeeGroup = objLoad.Data;
                    foreach (DSFeeSchedule.BasicFeeGroupRow dr in dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows)
                    {
                        //dr.BasicFeeGroupId = BasicFeeGroupId;
                        dr.ShortName = SearchedfieldsJSON["txtName"].Trim();
                        dr.Description = SearchedfieldsJSON["txtDescription"];
                        if (!string.IsNullOrEmpty(MDVUtility.ToStr(SearchedfieldsJSON["txtEndDate"])))
                            dr.EndDate = MDVUtility.ToDateTime(SearchedfieldsJSON["txtEndDate"]);
                        else
                            dr[dsBasicFeeGroup.BasicFeeGroup.EndDateColumn] = DBNull.Value;
                        if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlNextBasicFeeGroup"]))
                            dr.NextBasicFeeGroupId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlNextBasicFeeGroup"]);
                        else
                            dr.NextBasicFeeGroupId = MDVUtility.ToInt64(0);
                        dr.IsActive = MDVUtility.ToStr(SearchedfieldsJSON["chkIsActive"]) == "True" ? true : false;
                        //dr.CreatedBy = "";
                        //dr.CreatedOn = DateTime.Now;
                        dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                        dr.ModifiedOn = DateTime.Now;
                        if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlEntity"]))
                            dr.EntityId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlEntity"]);
                    }

                    #region Database Updation
                    //dsBasicFeeGroup.BasicFeeGroup.AddBasicFeeGroupRow(dr);
                    //dsBasicFeeGroup.BasicFeeGroup.AcceptChanges();

                    if (dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows.Count > 0)
                    {
                        //dsBasicFeeGroup.BasicFeeGroup.Rows[0].SetModified();
                        BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.UpdateBasicFeeGroup(ref dsBasicFeeGroup);
                        if (obj.Data != null)
                        {
                            var response = new
                            {
                                status = true,
                                message = Common.AppPrivileges.Update_Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Message
                            };
                            return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = ""
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    #endregion
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Fills the BasicFeeGroup.
        /// </summary>
        /// <param name="BasicFeeGroup">The Plan Fee Link identifier.</param>
        /// <returns>Json string containing key value pair or Exception message</returns>
        private string FillBasicFeeGroup(Int64 BasicFeeGroupId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Basic Fee Group", "VIEW")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    if (string.IsNullOrEmpty(MDVUtility.ToStr(BasicFeeGroupId)))
                    {
                        var response = new
                        {
                            status = false,
                            Message = MDVUtility.ToStr(Common.AppPrivileges.Select_Message)
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    else
                    {
                        DSFeeSchedule dsBasicFeeGroup = null;
                        BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.LoadBasicFeeGroup(BasicFeeGroupId, null, null, null, null);
                        if (obj.Data != null)
                        {
                            dsBasicFeeGroup = obj.Data;
                            if (dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows.Count > 0)
                            {
                                DataRow dr = dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows[0];
                                var keyValues = new Dictionary<string, string>
                        {
                          { "txtName", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.ShortNameColumn.ColumnName])},
                          { "txtDescription", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.DescriptionColumn.ColumnName])},
                          //{ "txtEndDate", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.EndDateColumn.ColumnName])},
                          { "txtEndDate", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.EndDateColumn.ColumnName]) !="" ? MDVUtility.ToDateTime(dr[dsBasicFeeGroup.BasicFeeGroup.EndDateColumn.ColumnName]).ToShortDateString():""},
                          { "ddlNextBasicFeeGroup", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.NextBasicFeeGroupIdColumn.ColumnName])},
                          { "ChkIsActive", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.IsActiveColumn.ColumnName])},
                          { "ddlEntity", MDVUtility.ToStr(dr[dsBasicFeeGroup.BasicFeeGroup.EntityIdColumn.ColumnName])}
                        };
                                System.Web.Script.Serialization.JavaScriptSerializer js = new System.Web.Script.Serialization.JavaScriptSerializer();
                                var response = new
                                {
                                    status = true,
                                    BasicFeeGroupFill_JSON = js.Serialize(keyValues)
                                };
                                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                            }
                            else
                            {

                                var response = new
                                {
                                    status = false,
                                    Message = AppPrivileges.No_Record_Message
                                };
                                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                            }
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }

                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }


        /// <summary>
        /// Fills the next basic fee group.
        /// </summary>
        /// <param name="EntityID">The entity identifier.</param>
        /// <returns>System.String.</returns>
        private string FillNextBasicFeeGroup(Int64 EntityID)
        {
            try
            {
                DSFeeSchedule dsBasicFeeGroup = null;
                BLObject<DSFeeSchedule> obj;
                obj = BLLFeeScheduleObj.LoadBasicFeeGroup(0, null, null, EntityID.ToString(), null);

                dsBasicFeeGroup = obj.Data;
                var response = new
                {
                    status = true,
                    BasicFeeGroupCount = dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows.Count,
                    NextBasicFeeGroupLoad_JSON = MDVUtility.JSON_DataTable(dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName]),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Deletes the BasicFeeGroup.
        /// </summary>
        /// <param name="ProcedureCategoryId">The Plan Fee Link identifier.</param>
        /// <returns>Json string containing Succes message or Exception message</returns>
        private string DeleteBasicFeeGroup(Int64 BasicFeeGroupId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Basic Fee Group", "DELETE")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    if (string.IsNullOrEmpty(MDVUtility.ToStr(BasicFeeGroupId)))
                    {
                        var response = new
                        {
                            status = false,
                            Message = MDVUtility.ToStr(Common.AppPrivileges.CheckBox_Message)
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    else
                    {
                        BLObject<string> obj = BLLFeeScheduleObj.DeleteBasicFeeGroup(MDVUtility.ToStr(BasicFeeGroupId));
                        if (obj.Data == "")
                        {
                            var response = new
                            {
                                status = true,
                                Message = Common.AppPrivileges.Delete_Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Data
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        private string UpdateBasicFeeGroupIsActive(Int64 BasicFeeGroupId, Int64 IsActive)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Basic Fee Group", "EDIT")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    DSFeeSchedule dsBasicFeeGroup = null;
                    BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.LoadBasicFeeGroup(BasicFeeGroupId, null, null, null, null);
                    dsBasicFeeGroup = obj.Data;
                    if (dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows.Count > 0)
                    {
                        DataRow dr = dsBasicFeeGroup.Tables[dsBasicFeeGroup.BasicFeeGroup.TableName].Rows[0];
                        dr[dsBasicFeeGroup.BasicFeeGroup.IsActiveColumn.ColumnName] = IsActive;

                        BLObject<DSFeeSchedule> objUser = BLLFeeScheduleObj.UpdateBasicFeeGroup(ref dsBasicFeeGroup);
                        string successMsg;
                        if (objUser.Data != null)
                        {
                            if (IsActive == 0)
                                successMsg = Common.AppPrivileges.Inactive_Message;
                            else
                                successMsg = Common.AppPrivileges.Active_Message;
                            var response = new
                            {
                                status = true,
                                message = successMsg
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = objUser.Message
                            };
                            return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }
        #endregion

        #region Service Command Handler
        /// <summary>
        /// Handle the Plan Fee Link Detail Commands and call to the respective methods.
        /// </summary>
        /// <param name="context">The context.</param>
        public void CommandHandler(HttpContext context)
        {
            string cammandAction = context.Request.QueryString["cammandAction"].ToUpper();

            switch (cammandAction)
            {
                case "SAVE_BASIC_FEE_GROUP":
                    {
                        string fieldsJSON = context.Request["BasicFeeGroupData"];
                        string strJSONData = SaveBasicFeeGroup(fieldsJSON);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "FILL_BASIC_FEE_GROUP":
                    {
                        string strBasicFeeGroupId = context.Request["BasicFeeGroupID"];
                        string strJSONData = FillBasicFeeGroup(MDVUtility.ToInt64(strBasicFeeGroupId));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "FILL_NEXT_BASIC_FEE_GROUP":
                    {
                        string strEntityId = context.Request["EntityID"];
                        string strJSONData = FillNextBasicFeeGroup(MDVUtility.ToInt64(strEntityId));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "DELETE_BASIC_FEE_GROUP":
                    {
                        string strBasicFeeGroupId = context.Request["BasicFeeGroupID"];
                        string strJSONData = DeleteBasicFeeGroup(MDVUtility.ToInt64(strBasicFeeGroupId));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "UPDATE_BASIC_FEE_GROUP":
                    {
                        string fieldsJSON = context.Request["BasicFeeGroupData"];
                        Int64 BasicFeeGroupID = MDVUtility.ToInt64(context.Request["BasicFeeGroupID"]);
                        string strJSONData = UpdateBasicFeeGroup(fieldsJSON, BasicFeeGroupID);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "UPDATE_BASIC_FEE_GROUP_ACTIVE_INACTIVE":
                    {
                        Int64 BasicFeeGroupID = MDVUtility.ToInt64(context.Request["BasicFeeGroupID"]);
                        Int64 IsActive = MDVUtility.ToInt64(context.Request["IsActive"]);
                        string strJSONData = UpdateBasicFeeGroupIsActive(BasicFeeGroupID, IsActive);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
            }
        }
        #endregion
    }
}